aleph/oauth.py

```diff
@@ -7,9 +7,14 @@
 from cryptography.x509 import load_pem_x509_certificate
 from cryptography.hazmat.backends import default_backend
 from servicelayer.extensions import get_entry_point
+from jose import jwt as jwt_jose
+from jose import jwk
+from jose.utils import base64url_decode
```

Comment:
I think we already have two implementations of JWT in use, I'd like to avoid a third. Can you check out https://docs.authlib.org/en/latest/jose/index.html instead? I'm pretty sure authlib also has solutions for some of the key parsing issues you're solving below.